### PR TITLE
[mongo] add tcmalloc metrics

### PR DESCRIPTION
**Pull request by yannmh**
Rebased of #1832 on master, thanks a lot @benmccann !

My two cents: 
Split the metric list in multiple parts:
* `COMMON_METRICS`: default metrics to collect
* `V2_ONLY_METRICS`: metrics available in MongoDB 2.x only
* `TCMALLOC_METRICS`: TCMalloc memory allocator metrics

It's easier to assess the check coverage in our CI, and gives more
control on the level of metric collection.

Create a `collect_tcmalloc_metrics` option (default to False).

### CODE REVIEW

File: checks.d/mongo.py [hunk 1]
```diff
@@ -6,124 +6,132 @@
...
+        "stats.indexes": GAUGE,
+        "stats.indexSize": GAUGE,
+        "stats.nsSizeMB": GAUGE,
+        "stats.numExtents": GAUGE,
+        "stats.objects": GAUGE,
+        "stats.storageSize": GAUGE,
+        "uptime": GAUGE,
+    }
+
+    V2_ONLY_METRICS = {
```
Comment (benmccann) [inline checks.d/mongo.py]: I don't think this is the entire list of metrics available only in mongo 2.x. E.g. I don't think that Mongo 3.x will return "cursors.totalOpen" or "cursors.timedOut"

File: checks.d/mongo.py [hunk 2]
```diff
@@ -200,11 +243,20 @@ def check(self, instance):
...
         db_name = parsed.get('database')
-        clean_server_name = server.replace(password, "*"*5) if password is not None else server
+        clean_server_name = server.replace(password, "*" * 5) if password is not None else server
 
         tags = instance.get('tags', [])
         tags.append('server:%s' % clean_server_name)
 
+        # Get the list of metrics to collect
+        collect_tcmalloc_metrics = _is_affirmative(
+            instance.get('collect_tcmalloc_metrics', False)
```
Comment (benmccann) [inline checks.d/mongo.py]: If we're going to add an option for collect_tcmalloc_metrics then we can use this to decide whether to request tcmalloc stats on older versions of MongoDB as well. They're starting to return them in 3.1.8 and later by default, but you can request them on older versions of MongoDB by passing `tcmalloc:true` when calling serverStats on line 316.

https://jira.mongodb.org/browse/SERVER-19941

File: checks.d/mongo.py (hunk 1 above)
Comment (yannmh) [inline checks.d/mongo.py]: I just ran some tests against a MongoDB 2.6 and 3.0 instances, the `V2_ONLY_METRICS ` seens exhaustive.
`cursors.totalOpen` or `cursors.timedOut` are still part of MongoDB 3.0 as stated in the [MongoDB 3.0 documentation] (https://docs.mongodb.org/manual/reference/command/serverStatus/#serverStatus.cursors.totalOpen)

**Comment by yannmh** [comment]
Thanks for your feedback @benmccann. I adressed your comments and rebased the PR.

It's ready to be merged now :ship:

File: checks.d/mongo.py (hunk 2 above)
Comment (yannmh) [inline checks.d/mongo.py]: That's good to know, thanks @benmccann. Added the change to my commit.

**Pull request merged**